
From: Coywolf Qi Hunt <coywolf@greatcn.org>

I just find a bug that ``make distclean'' cannot remove the editor backup
files and the like when using build directory.  That is because the find
command is improperly searching the build directory instead of the
$(srctree) it should.

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/Makefile |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

diff -puN Makefile~kbuild-distclean-srctree-fix Makefile
--- 25/Makefile~kbuild-distclean-srctree-fix	2004-06-23 21:44:36.473521016 -0700
+++ 25-akpm/Makefile	2004-06-23 21:44:36.477520408 -0700
@@ -866,7 +866,7 @@ $(clean-dirs):
 clean: archclean $(clean-dirs)
 	$(call cmd,rmdirs)
 	$(call cmd,rmfiles)
-	@find . $(RCS_FIND_IGNORE) \
+	 @find $(srctree) $(RCS_FIND_IGNORE) \
 	 	\( -name '*.[oas]' -o -name '*.ko' -o -name '.*.cmd' \
 		-o -name '.*.d' -o -name '.*.tmp' -o -name '*.mod.c' \) \
 		-type f -print | xargs rm -f
_
